### ISSUE TITLE
Specify custom loader columns by name

### ISSUE THREAD

**Issue by sean-lockwood** [spacetelescope/specviz#200]
Feature request, not a bug...

Rather than specifying a custom loader's column via numerical position, you could use the FITS column name.  Instead of:
```
dispersion:
  hdu: 1
  col: 2
```
It would be nice to specify:
```
dispersion:
  hdu: 1
  col: 'WAVELENGTH'
```
(Additionally, I believe the FITS standard makes column names case-insensitive.)

This is helpful because a loader would apply to more file types and require less code replication, assuming column names are the same.

**Comment by pllim** [comment]
Yeah, that is a reasonable request. Personally, not sure if this YAML thing is the way to go. General feedback we get is that it is hard to use. Perhaps we don't need this anymore if we switch to `specutils` in the future...

**Comment by nmearl** [comment]
> Perhaps we don't need this anymore if we switch to specutils in the future...

I'm not sure how `specutils` would offer anything remotely similar to custom loaders?

**Comment by nmearl** [comment]
The alternative would be to implement custom loaders in python, and have the user be responsible for constructing the `Data` object that then gets stored.

**Comment by pllim** [comment]
We might have to rethink this for long term support. This is a GUI application, not a universal table loader. The loader stuff should be somewhere else if it is reasonably complicated. At some point, we just have to draw the line and say something like, "If <third party package> cannot read this, you have to reformat your crazy data."

**Comment by pllim** [comment]
Anyways... tl;dr -- Good request *if* we decide to stick with custom YAML for every format out there.

**Comment by ibusko** [comment]
This issue is 2 years old. There seems to be now a good overlap in between the request / suggestion in here, and the spectrum ingestion wizard that was implemented later. I wonder if this could be just closed now.
